File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int a = m-1, b = n-1, k=m+n-1;

        while(a >= 0 && b >= 0 && k>=0) {
            if(nums1[a] > nums2[b]) {
                nums1[k] = nums1[a];
                a--;
            } else {
                nums1[k] = nums2[b];
                b--;
            }
            k--;
        }

        while(b >= 0 && k >= 0) {
            nums1[k] = nums2[b];
            b--;
            k--;
        }
       
        while(a >= 0 && k >= 0) {
            nums1[k] = nums1[a];
            a--;
            k--;
        }
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        a, b = m-1, n-1

        for i in range(a+b+1, -1, -1):
            if b >= 0 and a >= 0:
                if nums2[b] > nums1[a]:
                    nums1[i] = nums2[b]
                    b-=1
                else:
                    nums1[i] = nums1[a]
                    a-=1
            elif a >= 0:
                nums1[i] = nums1[a]
                a-=1
            elif b >= 0:
                nums1[i] = nums2[b]
                b-=1
